### DISCUSSION TITLE
RDF4J 4.0.1

### DISCUSSION THREAD

**Discussion by hmottestad** [eclipse-rdf4j/rdf4j#3901] [Announcements]
RDF4J 4.0.1 is now available. This is a patch release fixing 8 bugs.

For a complete overview, see [all issues fixed in 4.0.1](https://github.com/eclipse/rdf4j/milestone/79?closed=1).

<hr /><em>This discussion was created from the release <a href='https://github.com/eclipse/rdf4j/releases/tag/4.0.1'>4.0.1</a>.</em>

**Comment by naturzukunft** [comment]
Hi there, i have some strange behaviour upgrading to 4.0.1 from 3.7.4.
There are some rdf4j 3.7.2 dependencies and i currently did not understand from where they come.
E.g. rdf4j-model, rdf4j-rio-api, rdf4j-util

**Reply by naturzukunft** [reply]
Maybe it's an Eclipse problem!

**Reply by naturzukunft** [reply]
No, it's also on cmd-line. It's a runtmne Problem

**Reply by naturzukunft** [reply]
Today was one of that dependency hell days! I gave up. have to concentrate on my bugs ;-)

**Reply by frensjan** [reply]
Perhaps you can use `mvn dependency:tree` to check the origin of these dependencies.

I had a similar issue due to a dependency that had a dependency on 3.7.x. When updating my main project to 4.x, the `rdf4j-util` dependency was no longer managed by the `rdf4j-bom` (since 4.x no longer includes `rdf4j-util`). That shouldn't be the case for `rdf4j-model` though.
